### PR TITLE
Add more Witherden-Vincent rules

### PR DESCRIPTION
**Pull request by drwells**
I want to use the even rules since they are sometimes spaced more nicely.

I'm not sure whether it makes more sense to add the rules this way or to implement a second `QWitherdenVincentSimplexEven` class - in both cases `n_points_1D` is not that meaningful. I opted for this one since all these quadrature rules are in the same family, so they should be in the same class, but it might make sense to split them so that we can construct rules with a type and an integer.

### CODE REVIEW

**Comment by drwells** [comment]
@peterrum we could do the same thing for `QGaussSimplex` while implementing even-order rules: e.g., we really want to use a degree 2 rule for `TET4`. What do you think?

**Review by peterrum — APPROVED** [review]

**Comment by peterrum** [comment]
> we could do the same thing for QGaussSimplex while implementing even-order rules: e.g., we really want to use a degree 2 rule for TET4. What do you think?

Good idea!

File: include/deal.II/base/quadrature_lib.h [hunk 1]
```diff
@@ -855,7 +865,8 @@ class QWitherdenVincentSimplex : public QSimplex<dim>
    * Constructor taking the number of quadrature points in 1D direction
    * @p n_points_1D.
```
Comment (kronbichler) [inline include/deal.II/base/quadrature_lib.h:866]: Would you mind removing this comment (as one needs the full information in the class definition in my opinion) and simply state
```suggestion
   * Constructor. See the general class description for details on the parameters.
```

**Review by kronbichler — APPROVED** [review]
Looks good apart from the comment.

**Comment by drwells** [comment]
Fixed!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**